Tasks/AzureResourceManagerTemplateDeploymentV3/models/ARMDeployTaskParameters.ts

```diff
@@ -0,0 +1,112 @@
+import tl = require("azure-pipelines-task-lib/task");
+import msRestAzure = require('azure-arm-rest-v2/azure-arm-common');
+import { AzureRMEndpoint } from 'azure-arm-rest-v2/azure-arm-endpoint';
+
+export class ARMDeployTaskParameters {
+
+    public action: string;
+    public resourceGroupName: string;
+    public location: string;
+    public csmFile: string;
+    public csmParametersFile: string;
+    public templateLocation: string;
+    public csmFileLink: string;
+    public csmParametersFileLink: string;
+    public overrideParameters: string;
+    public outputVariable: string;
+    public subscriptionId: string;
+    public endpointPortalUrl: string;
+    public deploymentName: string;
+    public deploymentMode: string;
+    public credentials: msRestAzure.ApplicationTokenCredentials;
+    public deploymentOutputs: string;
+    public addSpnToEnvironment: boolean;
+    public connectedService: string;
+    public deploymentScope: string;
+    public managementGroupId: string;
+
+    private async getARMCredentials(connectedService: string): Promise<msRestAzure.ApplicationTokenCredentials> {
+        var azureEndpoint = await new AzureRMEndpoint(connectedService).getEndpoint();
+        return azureEndpoint.applicationTokenCredentials;
+    }
+
+    public async getARMTemplateDeploymentTaskParameters() : Promise<ARMDeployTaskParameters>
+    {
+        try {
+
+            //Deployment Scope
+            this.deploymentScope = tl.getInput("deploymentScope");
+            if(!this.deploymentScope){
+                this.deploymentScope = "Resource Group";
+            }
+
+            var resourceGroupNameInServiceConnection;
+
+            //Service Connection
+            this.connectedService = tl.getInput("ConnectedServiceName", true);
+            var endpointTelemetry = '{"endpointId":"' + this.connectedService + '"}';
+            console.log("##vso[telemetry.publish area=TaskEndpointId;feature=AzureResourceManagerTemplateDeployment]" + endpointTelemetry);
+            this.endpointPortalUrl = tl.getEndpointDataParameter(this.connectedService, "armManagementPortalUrl", true);
+            var armServiceConnectionScope = tl.getEndpointDataParameter(this.connectedService, 'ScopeLevel', true);
+            if(!!armServiceConnectionScope && armServiceConnectionScope === "Subscription"){
+                var armServiceConnectionAuthScope = tl.getEndpointAuthorizationParameter(this.connectedService, 'scope', true);
+                if(!!armServiceConnectionAuthScope){
+                    var armServiceConnectionAuthScopeSplit = armServiceConnectionAuthScope.split("/");
+                    if(!!armServiceConnectionAuthScopeSplit[4]){
```

Comment:
this can fail in case scope does not contain 4 parts. you should check for the length of array first

Reply:
No, this doesn't fails. The double exclamation takes care of this.